module/VuFind/src/VuFind/Auth/Shibboleth/ShibbolethFactory.php

```diff
@@ -60,8 +62,17 @@ public function __invoke(ContainerInterface $container, $requestedName,
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
+        $config = $container->get(\VuFind\Config\PluginManager::class)->get('config');
+        $override = $config->Shibboleth->override;
+        $loader = null;
+        if (!empty($override)) {
```

Comment:
Might be worth moving this logic to a protected getConfigurationLoader() method of the factory, so it's easier to override with a custom loader by extending the factory.